cocotb/share/lib/fli/FliImpl.h

```diff
@@ -373,7 +373,7 @@ class FliRealObjHdl : public FliValueObjHdl {
 
     ~FliRealObjHdl() override {
         if (m_mti_buff != NULL)
-            free(m_mti_buff);
+            delete m_mti_buff;
```

Comment:
An even nicer option would be to change the type of `m_mti_buff` from `T*` to `std::unique_ptr<T[]>`, which will handle this delete for you.

Fine to leave this for another patch, but you either need to fix these to `delete[]`, or just remove them entirely.

Reply:
I think I might, it's been so long since I've used `new` and `delete` raw that I apparently forgot how to use them correctly, might as well take that out of the equation.